src/darker/__main__.py

```diff
@@ -572,9 +573,12 @@ def main(  # pylint: disable=too-many-locals,too-many-branches,too-many-statemen
     else:
         # In other modes, only reformat files which have been modified.
         if git_is_repository(root):
-            changed_files_to_reformat = git_get_modified_python_files(
-                files_to_process, revrange, root
-            )
+            # Get the modified files only.
+            repo_root = git_get_repo_root(root)
+            modified_files = git_get_modified_python_files(paths, revrange, repo_root)
```

Comment:
```suggestion
            modified_files = git_get_modified_python_files(files_to_process, revrange, repo_root)
```
Shouldn't this use `files_to_process` to make sure files excluded by Black by default don't get processed?

Reply:
The problem with that is it would cause the same issue again if the command being too long. 
Taking your correction, what it should do is get all changed files and then union with `files_to_process` to make sure only files that are in the latter are kept.

Reply:
> get all changed files and then union with `files_to_process`

You probably mean intersect instead of union. But yes, this would probably be a good solution.

Can you think of a way to create a test case for this? Generate a large enough number of files in a Git repository and run `main()` on it?

You'll find some tests in `test_command_line.py` and `test_main.py` with examples of helpers to make things happen for such cases.

Reply:
Yeah, sorry, intersect :)

I was going to create a git repo with enough files with a given filename length that exceeds the OS maximum. It's a bit slow, but I can't think of a better way atm to reliably test this

Reply:
The number of files could be reduced by nesting them in a deep file hierarchy with long directory names.